firebase_admin/db.py

```diff
@@ -0,0 +1,641 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Firebase Database module.
+
+This module contains functions and classes that facilitate interacting with the Firebase database.
+It supports basic data manipulation operations, as well as a complex queries such as limit queries,
+and range queries. This module uses the Firebase REST API underneath. Therefore it does not
+support realtime update notifications.
+"""
+
+import collections
+import json
+import numbers
+
+import requests
+import six
+from six.moves import urllib
+
+from firebase_admin import utils
+
+_DB_ATTRIBUTE = '_database'
+_INVALID_PATH_CHARACTERS = r'[].#$'
+_RESERVED_FILTERS = ('$key', '$value', '$priority')
+
+
+def get_reference(path='/', app=None):
+    """Returns a database Reference representing the node at the specified path.
+
+    If no path is specified, this function returns a Reference that represents the database root.
+
+    Args:
+      path: Path to a node in the Firebase realtime database (optional).
+      app: An App instance (optional).
+
+    Returns:
+      Reference: A newly initialized Reference.
+
+    Raises:
+      ValueError: If the specified path or app is invalid.
+    """
+    client = utils.get_app_service(app, _DB_ATTRIBUTE, _Client.from_app)
+    return Reference(client=client, path=path)
+
+def _parse_path(path):
+    """Parses a path string into a set of segments."""
+    if not isinstance(path, six.string_types):
+        raise ValueError('Invalid path: "{0}". Path must be a string.'.format(path))
+    if any(ch in path for ch in _INVALID_PATH_CHARACTERS):
+        raise ValueError(
+            'Invalid path: "{0}". Path contains illegal characters.'.format(path))
+    segments = []
```

Comment:
You could replace this loop/append structure with the following if you were so inclined:

```python
return [seg for seg in path.split('/') if seg]
```